fili-core/src/test/groovy/com/yahoo/bard/webservice/data/config/table/BaseTableLoaderSpec.groovy

```diff
@@ -8,95 +8,269 @@ import com.yahoo.bard.webservice.data.config.ResourceDictionaries
+        physDefs = TestPhysicalTableDefinitionUtils.buildShapeTableDefinitions(new TestDimensions(), metricNames)
         dimNames = TestApiDimensionName.getByLogicalTable(SHAPES)
+        tableNames = physDefs.stream().map({it -> it.getName()}).collect(Collectors.toSet())
 
         dims = dimNames.collect {
             name -> new KeyValueStoreDimension(TestDimensions.buildStandardDimensionConfig(name))
         }
         dicts.getDimensionDictionary().addAll(dims)
+
+        physicalTableSchema = Mock(PhysicalTableSchema)
+        physicalTableSchema.getColumns(DimensionColumn.class) >> []
+
+        physicalTable = Mock(PhysicalTable)
+        physicalTable.getTableName() >> TableName.of('definition2')
+        physicalTable.getSchema() >> physicalTableSchema
+
+        definition1 = new SimpleDependencyPhysicalTableDefinition('definition1', 'definition2')
+        definition2 = new SimpleDependencyPhysicalTableDefinition('definition2', physicalTable)
+        definition3 = new SimpleDependencyPhysicalTableDefinition('definition3', 'definition1')
+        definition4 = new SimpleDependencyPhysicalTableDefinition('definition4', 'definition4')
+        definition5 = new SimpleDependencyPhysicalTableDefinition('definition5', 'definition6')
+        definition6 = new SimpleDependencyPhysicalTableDefinition('definition6', 'definition5')
+
+        allDependentDefinitions = [definition1, definition2, definition3, definition4, definition5, definition6]
+
     }
 
     def "table group has correct contents after being build"() {
         when:
         TableGroup group = loader.buildDimensionSpanningTableGroup(
-                apiNames,
-                metricNames,
+                tableNames,
                 physDefs,
-                dicts
+                dicts,
+                apiNames
         )
 
         then:
         group.physicalTables.size() == physDefs.size()
         group.apiMetricNames == apiNames
     }
 
-    def "loading distinct physical tables results in correct tables in dictionary"() {
-        setup:
-        List<PhysicalTableDefinition> allDefs = physDefs as List
 
+    def "loading distinct physical tables without dependency results in correct tables in dictionary and table group"() {
         when:
-        PhysicalTable t1 = loader.loadPhysicalTable(allDefs[0], metricNames, dicts)
-        PhysicalTable t2 = loader.loadPhysicalTable(allDefs[1], metricNames, dicts)
+        TableGroup group = loader.buildDimensionSpanningTableGroup(
+                tableNames,
+                physDefs,
+                dicts,
+                apiNames
+        )
 
         then:
-        dicts.physicalDictionary.size() == 2
-        dicts.physicalDictionary.containsValue(t1)
-        dicts.physicalDictionary.containsValue(t2)
+        dicts.physicalDictionary.size() == 6
+        group.physicalTables.stream().map({it.getTableName()}).collect(Collectors.toSet()) as Set == tableNames
```

Comment:
could be groovier: `group.physicalTables.collect {it.tableName} as Set`

<sub>Note: Applies elsewhere in PR</sub>